File: src/main/java/com/und/security/repository/UserRepository.java
package com.und.security.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import com.und.model.security.User;

/**
 * Created by shiv on 21/07/17.
 */
public interface UserRepository extends JpaRepository<User, Long> {
    User findByUsername(String username);
}

File: src/main/kotlin/com/und/security/repository/UserRepository.kt
package com.und.security.repository

import org.springframework.data.jpa.repository.JpaRepository
import com.und.security.model.User

/**
 * Created by shiv on 21/07/17.
 */
interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String): User?
}
